src/ring_theory/algebra.lean

```diff
@@ -153,6 +154,14 @@ instance module.endomorphism_algebra (R : Type u) (M : Type v)
   commutes' := by intros; ext; simp,
   smul_def' := by intros; ext; simp }
 
+set_option class.instance_max_depth 50
+instance matrix_algebra (n : ℕ) (R : Type u)
```

Comment:
I think you can do this for arbitrary `fintypes`. You don't need `fin n`.

Reply:
Agreed